### ISSUE TITLE
Missing components on LG A/C boards for CN-Remo connector – Need help identifying parts

### ISSUE THREAD

**Issue by fx-host-assistant** [JanM321/esphome-lg-controller#81]
Hello everyone,
I'm working with LG A/C boards that have the CN-Remo connector unsoldered (they are a bit old), and I noticed that they’re also missing a resistor, a diode, and a capacitor between the connector and the data line to the A/C unit. It seems that this layout might be the same across similar LG boards, so I’m hoping someone could check theirs and help me identify the missing components.

Here’s what I’ve gathered so far:

    Resistor: 12k Ohm
    Diode: Likely a Small Signal Fast Switching Diode (TH, maybe DO-35)
    Capacitor: Unknown type and value

If anyone has information on the specific capacitor value and any additional details about these components, I’d be very grateful for your help. Thank you!

**Comment by PVi1** [comment]
Hi @Smandurlo 
here are photos of those boards. One is without cnremo and the other is from s12et.nsj with cnremo embedded.
the question is if there isn’t missing code in the eeprom, too.
https://share.icloud.com/photos/083Q5lFE1z2m-IWrmDbWEikww

I my case the board without cnremo does not have wifi(cnlink) port populated too.

**Comment by Smandurlo** [comment]
Hi @PVi1 thank you very much for your pics! Something missing in the eeprom is a possibility, I hope it won't be the case :-)

**Comment by PVi1** [comment]
Hi @Smandurlo,
any news?

**Comment by Smandurlo** [comment]
> Hi [@Smandurlo](https://github.com/Smandurlo), any news?

Hey, thanks for asking! I was waiting for the European spring to work on the PCBs, but in the meantime, the external compressor broke down two weeks ago, and the damage isn’t worth repairing :-( we will never know it........

**Comment by lagomCat** [comment]
![Image](https://github.com/user-attachments/assets/c408388c-47e8-4936-a01a-bcf5b28fabc7)

**Comment by Smandurlo** [comment]
it makes sense, but maybe there is another voltage divider on the esp module?

**Comment by lagomCat** [comment]
> это имеет смысл, но может быть на модуле esp есть еще один делитель напряжения?

I've already figured it out and realized that in this project, the device doesn't receive any data from the air conditioner's main unit — it only sends data periodically. So there's no feedback in this case. That means there's no point in changing the resistor value — communication over the bidirectional bus is currently only one-way.

If the project is to be further developed, it's possible to build a level-shifting circuit on the ESP32 side similar to the one on the air conditioner's board. That would give us full two-way communication.

But then, who’s going to write the code to handle the incoming data from the air conditioner and display it in the graphical interface? 😄

**Comment by Smandurlo** [comment]
Oh wow, so it is basically a remote without feedback? I thought it was bidirectional! It is quite disappointing... If someone changes the settings with the normal ir remote, it won't be changed in esphome/homeassistant

**Comment by JanM321** [comment]
> I've already figured it out and realized that in this project, the device doesn't receive any data from the air conditioner's main unit — it only sends data periodically. So there's no feedback in this case. 

That's incorrect, there's definitely two way communication. The controller receives settings and status information from the AC.

> If someone changes the settings with the normal ir remote, it won't be changed in esphome/homeassistant

Settings will be changed in HA if you change them using the ir remote or ThinQ.

**Comment by lagomCat** [comment]
> That's incorrect, there's definitely two way communication. The controller receives settings and status information from the AC.

However, in this particular hardware implementation, when sending messages from the ESP32 side, you effectively have a voltage divider made of 1 kΩ and 12 kΩ resistors (sometimes 4.7 kΩ depending on the air conditioner model).
If you claim there is feedback, then I’m confused — because that would go against the laws of physics. 😊

**Comment by hovaks** [comment]
> I ran into exactly the same issue with an LG S09EQR air conditioner. The circuitry for the CN_REMO connector was not populated on the PCB. I found the missing components in a service manual and soldered them onto the board myself. Now I’m waiting for an ESP32 module to arrive by mail. It would be really frustrating if, in addition to the unpopulated components, the engineers also disabled the serial port in firmware.
> 
> Also, I have a question about the circuit. There's a 12kΩ resistor shown in the schematic. In the LIN driver part (TJA1027) used in this project, the LIN line is pulled up to +12V via a 1kΩ resistor. That means when the AC controller drives its Tx pin low, the line is pulled down through a voltage divider made of 1kΩ and 12kΩ. But that would only drop the voltage on the LIN line to around 11.1V — far too high to be interpreted as a logical LOW by the LIN transceiver, according to the spec. I don’t understand how this is supposed to work. Wouldn’t it make more sense to increase the value of the pull-up resistor on the LIN side to get a proper voltage drop?
> 
> ![Image](https://github.com/user-attachments/assets/e801d317-2a5e-45cf-85e5-ad38d53c7e7c)
> 
> ![Image](https://github.com/user-attachments/assets/28d68799-6862-450a-bc94-89e838d3d655)
> 
> ![Image](https://github.com/user-attachments/assets/0d786cc1-815d-4ec2-9a67-fef8a3a01529)

Did you try, what was the situation with the firmware?
